pallets/eth-app/src/lib.rs

```diff
@@ -86,11 +89,21 @@ decl_module! {
             let who = ensure_signed(origin)?;
             let asset_id: BridgedAssetId = H160::zero();
 
-            // <asset::Module<T>>::do_burn(asset_id, &who, amount)?;
             let asset_id = <asset::Module<T>>::get_native_asset_id(asset_id);
-            let result = <assets::Module<T>>::assets_burn(&asset_id, &who, &amount.low_u128().saturated_into::<T::Balance>());
 
-            ensure!(result.is_ok(), Error::<T>::BurnFailure);
+            ensure!(T::Currency::can_slash(
```

Comment:
I think we also need to check ensure_can_withdraw, as I believe that when burning frozen amount should be considered.